Added Behat integration

wouterj opened the pull request:
This PR adds the necessary classes to integrate with Behat, similar to the existing PHPunit integration. I hope this is something you want to add, it would be of great help for our test suite :)

If you want this, do you want me to add a Behat test set-up to functionally test the integration?

dmaicher reviewed (commented):
Thanks for proposing this @wouterj :+1: 

Having this covered by a functional test would be nice. Will this require a lot of code?

wouterj commented on README.md:
```diff
@@ -59,11 +61,24 @@ It also includes a `StaticArrayCache` that will be automatically configured as m
+
+#### Using the Bundle with Behat
+
+Enable the extension in your Behat config (e.g. `behat.yml`)
```
Good points! I'll update it.

(and oh, I completely missed that part of the README, guess I should read the complete README next time :see_no_evil: )

wouterj commented on composer.json:
```diff
@@ -17,11 +17,15 @@
     "phpstan/phpstan": "^0.12"
   },
+  "conflict": {
+    "behat/behat": "<3.0"
```
Yes, indeed. I'll remove it (Behat 3 is out for so long that I think everyone actively working on their test suite is already on it).

wouterj commented:
> Having this covered by a functional test would be nice. Will this require a lot of code?

I'll have a look, I think it'll have the same or less code as the PHPunit functional test.

Thanks for your review, I'll try to find some time this week to work on this PR.

wouterj commented:
Updated the PR, I've added functional behat tests (similar to those existing in PHPunit). I couldn't understand what `PhpunitTest::testRollBackChangesWithReOpenedConnection` was testing, so I haven't copied this one to the Behat tests. I also didn't move the `testPreviousChangesAreRolledBack` tests to Behat, it's a bit non-behat-ish to do so. As all scenarios start with `Given there are 0 rows`, we already test it enough imho.

dmaicher reviewed (approved):
Nice :+1: Thanks

dmaicher commented:
@mnapoli @dkarlovi what do you think about this? Any chance you could review the behat part? Since you commented [here](https://github.com/dmaicher/doctrine-test-bundle/issues/48) I guess you are using this bundle with behat? :blush:

dmaicher commented:
@wouterj I added some changes on master (php8 support for example). Could you solve the conflicts please? I would like to merge this as well for release 6.4.0

dmaicher reviewed (approved):
Thanks @wouterj :+1:

wouterj commented:
Thank you for your time reviewing & merging this PR @dmaicher!
